fmlloader/src/main/java/net/minecraftforge/fml/loading/moddiscovery/AbstractModProvider.java

```diff
@@ -61,10 +61,6 @@ protected IModLocator.ModFileOrException createMod(Path path, boolean ignoreUnkn
         if (sj.moduleDataProvider().findFile(MODS_TOML).isPresent()) {
             LOGGER.debug(LogMarkers.SCAN, "Found {} mod of type {}: {}", MODS_TOML, type, path);
             mod = new ModFile(sj, this, ModFileParser::modsTomlParser);
-            if (mod.getModFileInfo().getFileProperties().containsKey(ModFileInfo.NOT_A_FORGE_MOD_PROP)) {
-                LOGGER.error(LogMarkers.SCAN, "Unable to load file \"{}\" because its mods.toml is requesting an invalid javafml loaderVersion (use \"*\" if you want to allow all versions) and is missing a forge modId dependency declaration (see the sample mods.toml in the MDK).", path);
```

Comment:
What happens when someone shoves a mod that was detected by this, for an old version (1.20.1?), into the new version?

Reply:
I thought the issue was that they changed the mods.toml format, which caused the ModContainer {or whatever it was that read things like the license and dependencies} to error out because of them making non-optional entries optional.

I'd keep it in place, if you feel like cleaning it up I dont really care.

Reply:
Can you post a example? Just for clarity. Because like I said I thought thr error was a NPE and not a "bad mod" or something.